src/snyk/snykCode/falsePositive/falsePositive.ts

```diff
@@ -0,0 +1,61 @@
+import { Marker } from '@snyk/code-client';
+import path from 'path';
+import { IVSCodeWorkspace } from '../../common/vscode/workspace';
+
+export class FalsePositive {
+  readonly files: Set<string>;
+
+  constructor(private workspace: IVSCodeWorkspace, suggestionFilePath: string, markers: Marker[]) {
+    const markerPositions = markers.flatMap(marker => marker.pos);
+    const filesArray = markerPositions.map(markerPosition =>
+      this.getAbsoluteMarkerFilePath(markerPosition.file, suggestionFilePath),
+    );
+
+    this.files = new Set(filesArray);
+  }
+
+  getAbsoluteMarkerFilePath(filePath: string, suggestionFilePath: string): string {
+    if (!filePath) {
+      // If no filePath reported, use suggestion file path as marker's path. Suggestion path is always absolute.
+      return suggestionFilePath;
+    }
+
+    const workspaceFolders = this.workspace.getWorkspaceFolders();
+    if (workspaceFolders.length > 1) {
+      return filePath;
+    }
+
+    // The Snyk Code analysis reported marker path is relative when in workspace with a single folder, thus need to convert to an absolute
+    return path.resolve(workspaceFolders[0], filePath);
+  }
+
+  /**
+   * May throw an error if file cannot be resolved.
+   */
+  async getContent(): Promise<string> {
+    let content = this.getMainHeader();
+
+    for await (const file of this.files) {
+      const doc = await this.workspace.openFileTextDocument(file);
+      content += this.appendFileHeader(doc.getText(), file);
```

Comment:
Any chance to set the type here to get syntax highlighting?

Reply:
It actually infers the type as a string since we set it to this type on line 36.